### ISSUE TITLE
spring-boot-starter-parent pom missing version for git-commit-id-plugin

### ISSUE THREAD

**Issue by shoffmeister** [spring-projects/spring-boot#16675]
https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-starters/spring-boot-starter-parent/pom.xml#L138

misses version information in pluginManagement for the plugin
```
  <groupId>pl.project13.maven</groupId>
  <artifactId>git-commit-id-plugin</artifactId>
```

This makes it impossible to consume that plugin in projects by _just_ referring to the plugin _managed_ by the Spring Boot parent. Instead, child projects need to explicitly include the version tag themselves, i.e. instead of being able to consume
```
  <groupId>pl.project13.maven</groupId>
  <artifactId>git-commit-id-plugin</artifactId>
```
these projects needs to consume using
```
  <groupId>pl.project13.maven</groupId>
  <artifactId>git-commit-id-plugin</artifactId>
  <version>${git-commit-id-plugin.version}</version>
```
See https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-dependencies/pom.xml#L3375 for the dependency management part.

So, in short: pom.xml of spring-boot-starter-parent should include
```
  <version>${git-commit-id-plugin.version}</version>
```
on the relevant plugin.

**Comment by shoffmeister** [comment]
With the way things are at the moment, simply consuming the plugin in build results in
```
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for <myproject>
[WARNING] 'build.plugins.plugin.version' for pl.project13.maven:git-commit-id-plugin is missing. @ <myproject>/pom.xml, line 294, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
```

**Comment by shoffmeister** [comment]
I am not familiar with the intricacies of Spring Boot, but it seems as if the specific pom mentioned here is the only one requiring fixing:

https://github.com/spring-projects/spring-boot/search?q=project13&unscoped_q=project13

**Comment by wilkinsona** [comment]
It sounds like you project isn't using Spring Boot's starter parent. In the case you will need to manage the version yourself. The current arrangement works for me when using `spring-boot-starter-parent` as the project's parent and declaring the plugin without a version:

```
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.1.4.RELEASE</version>
	</parent>
 	
	<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
			</plugin>
			<plugin>
				<groupId>pl.project13.maven</groupId>
				<artifactId>git-commit-id-plugin</artifactId>
			</plugin>
		</plugins>
	</build>
```

If this doesn't work for you and you'd like us to spend some more time investigating please take the time to provide a complete and minimal sample that reproduces the problem.

**Comment by shoffmeister** [comment]
My apologies - this is not Spring Boot parent pom _straight_ but an "uber pom" generated from that very parent pom by
```
                        <groupId>com.igormaznitsa</groupId>
                        <artifactId>uber-pom</artifactId>
```
This uber pom basically takes the Spring Boot parent pom, flattens the dependencies, and swaps the parent of the Spring Boot pom out for another (version information dependency management) pom.  

In the end, the "sort of Spring Boot parent uber pom" has the following lines
```
        <plugin>
          <groupId>pl.project13.maven</groupId>
          <artifactId>git-commit-id-plugin</artifactId>
          <executions>
...
```
which is exactly what the Spring Boot parent pom has.

I tried to reproduce without the uber pom and was not able to see the warning about build stability with the straight Spring Boot parent pom.
I will try to work with the author of the uber pom to determine the root cause of the problem.

**Comment by wilkinsona** [comment]
Thanks for the additional information. This sound like a problem with the `uber-pom` rather than Spring Boot so I'm going to close this issue. If you discover a problem with Spring Boot when working with the uber pom's author, please comment here and we can re-open this issue.
